cgmanifests/generate_cgmanifest.py

```diff
@@ -15,6 +33,59 @@
 
 registrations = []
 
+
+@dataclass(frozen=True)
+class GitDep:
+    commit: str
+    url: str
+
+
+git_deps = {}
+
+
+def add_github_dep(name, parsed_url):
+    segments = parsed_url.path.split("/")
+    org_name = segments[1]
+    repo_name = segments[2]
+    if segments[3] != "archive":
+        print("unrecognized github url path:" + parsed_url.path)
+        return
+    git_repo_url = "https://github.com/%s/%s.git" % (org_name, repo_name)
+    if len(segments) == 5 and len(PurePosixPath(segments[4]).stem) == 40:
+        commit = PurePosixPath(segments[4]).stem
```

Comment:
if segments is the result of splitting on '/', why do we need to convert each one further with PurePosixPath().stem?

Reply:
Remove file extension.  segments[4] is like 3c73d91c0b04e2b59462f0a741be8c07024c1bc0.zip